ntbs-service/Pages/Notifications/Edit/PatientDetails.cshtml.cs

```diff
@@ -111,16 +115,21 @@ protected override async Task<IActionResult> PrepareAndDisplayPageAsync(bool isB
                 return null;
             }
 
-            var notificationIds = await NotificationRepository.GetNotificationIdsByNhsNumberAsync(nhsNumber);
+            var ntbsNhsNumberMatches = await NotificationRepository.GetNotificationIdsByNhsNumberAsync(nhsNumber);
             var idsInGroup = group?.Notifications?.Select(n => n.NotificationId) ?? new List<int>();
-            var filteredIds = notificationIds
+            var filteredNtbsIdDictionary = ntbsNhsNumberMatches
                 .Except(idsInGroup)
                 .Where(n => n != NotificationId)
                 .ToDictionary(
                     id => id.ToString(),
                     id => RouteHelper.GetNotificationPath(id, NotificationSubPaths.Overview));
 
-            return filteredIds;
+            var legacyIds = await _migrationRepository.GetLegacyNotificationNhsNumberMatches(nhsNumber);
```

Comment:
I think `legacyNhsNumberMatches` would be a more consistent name (in the context of the other names in this method.

Reply:
Also, I think it would be a good idea to make these requests in parallel - since the migration repository isn't part of the EF context there shouldn't be any issue with doing so and it would save a bit of time.